ModsenOnlineStore.Store.Application/Services/CommentServices/CommentService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using AutoMapper;
+using ModsenOnlineStore.Common;
+using ModsenOnlineStore.Store.Application.Interfaces.CommentInterfaces;
+using ModsenOnlineStore.Store.Domain.DTOs.CommentDTOs;
+using ModsenOnlineStore.Store.Domain.Entities;
+
+namespace ModsenOnlineStore.Store.Application.Services.CommentServices
+{
+    public class CommentService : ICommentService
+    {
+        private readonly ICommentRepository commentRepository;
+        private readonly IMapper mapper;
+
+        public CommentService(ICommentRepository commentRepository, IMapper mapper)
+        {
+            this.commentRepository = commentRepository;
+            this.mapper = mapper;
+        }
+
+        public async Task<ResponseInfo<List<GetCommentDto>>> GetAllComments()
+        {
+            var comments = await commentRepository.GetAllComments();
+            var commentDtos = comments.Select(mapper.Map<GetCommentDto>).ToList();
+
+            return new ResponseInfo<List<GetCommentDto>>(data: commentDtos, success: true, message: "all comments");
+        }
+
+        public async Task<ResponseInfo<GetCommentDto>> GetCommentById(int id)
+        {
+            var comment = await commentRepository.GetCommentById(id);
+
+            if (comment is null)
+            {
+                return new ResponseInfo<GetCommentDto>(data: null, success: true, message: "comment");
+            }
+
+            var commentDto = mapper.Map<GetCommentDto>(comment);
+
+            return new ResponseInfo<GetCommentDto>(data: commentDto, success: true, message: "comment");
+        }
+
+        public async Task<ResponseInfo<string>> AddComment(AddCommentDto addCommentDto)
+        {
+            var comment = mapper.Map<Comment>(addCommentDto);
+            await commentRepository.AddComment(comment);
+
+            return new ResponseInfo<string>(data: "added successfully", success: true, message: "comment");
+        }
+
+        public async Task<ResponseInfo<string>> UpdateComment(UpdateCommentDto updateCommentDto)
+        {
+            var comment = mapper.Map<Comment>(updateCommentDto);
+            await commentRepository.UpdateComment(comment);
+
+            return new ResponseInfo<string>(data: "updated successfully", success: true, message: "comment");
+        }
+
+        public async Task<ResponseInfo<string>> RemoveCommentById(int id)
+        {
+            await commentRepository.RemoveCommentById(id);
+
+            return new ResponseInfo<string>(data: "removed successfully", success: true, message: "comment");
```

Comment:
what is the purpose of such data and message
`success: true` already tells us that object was deleted
`message` it's a bit of an abstract name: is it failure message or some other kind of message?
make another ResponseInfo class without any data and non-generic so it store success flag and message

Reply:
NoDataResponseInfo class was created, now success flag can be true or false according to the result of operation, message tells that operation done or purpose of operation failure

Reply:
It is already clear that operation is done when `success = true`. So I think that the only one reason to have this message is failure message